app/controllers/ex_links_controller.rb

```diff
@@ -0,0 +1,112 @@
+class ExLinksController < ApplicationController
+  before_action :set_ex_link, only: [:show, :edit, :update, :destroy]
+
+  # GET /ex_links
+  def index
+    if params[:tag]
+      @ex_links = get_exlinks_if_all_tags_present(params[:tag])
+    else
+      @ex_links = ExLink.all
+    end
+  end
+
+  # GET /ex_links/1
+  def show
+  end
+
+  # GET /ex_links/new
+  def new
+    @ex_link = ExLink.new
+  end
+
+  # GET /ex_links/1/edit
+  def edit
+  end
+
+  # POST /ex_links
+  def create
+    @ex_link = ExLink.new(ex_link_params)
+    if @ex_link.save
+      redirect_to @ex_link, notice: 'Ex link was successfully created.'
+    else
+      render :new
+    end
+  end
+
+  # PATCH/PUT /ex_links/1
+  def update
+    if @ex_link.update(ex_link_params)
+      redirect_to @ex_link, notice: 'Ex link was successfully updated.'
+    else
+      render :edit
+    end
+  end
+
+  # DELETE /ex_links/1
+  def destroy
+    @ex_link.destroy
+    redirect_to ex_links_url, notice: 'Ex link was successfully destroyed.'
```

Comment:
I'm using some method that I put in `application_controller` which is called as:
  - `notice: alert_create(ExLink)`
  - `notice: alert_update(ExLink)`
  - `notice: alert_destroy(ExLink)`

For giving translated notices.